### DISCUSSION TITLE
next.config.js executes multiple times?

### DISCUSSION THREAD

**Discussion by paul-vd** [vercel/next.js#11863] [Help]
Hi,

So with my current setup, I noticed that the in dev mode `next dev` the `next.config.js` export runs twice.
![image](https://user-images.githubusercontent.com/39598117/79167116-ff748d80-7de6-11ea-9dc7-69b075413256.png)
The `isDev, isProd, isStaging`  console log happens within the building phase and I would expect it to run once.

I never experienced this with the previous versions.

Here is the example of my current next.config.js: https://github.com/PaulPCIO/nextjs-boilerplate-ts/blob/master/next.config.js

**Comment by williamhaley** [comment]
I noticed this today as well with `next@14.0.3` for a `dev` build. I see `next.config.js` evaluated two times.

Towards the top of `next.config.js` I have a `console.log(process.argv)`. See how the two examples here differ.

# First iteration

```
[
  '/Users/will.haley/.nodenv/versions/18.18.2/bin/node',
  '/Users/will.haley/src/my_app_redacted/node_modules/.bin/next',
  'dev',
  '-p',
  '8082'
]
```

# Second iteration

```
[
  '/Users/will.haley/.nodenv/versions/18.18.2/bin/node',
  '/Users/will.haley/src/my_app_redacted/node_modules/next/dist/server/lib/start-server.js'
]
```

Not sure why, but I'm assuming `loadConfig` gets invoked multiple times between these two iterations. https://github.com/vercel/next.js/blob/v14.0.3/packages/next/src/server/config.ts#L888C31-L888C41

**Comment by canmeng568** [comment]
how to solve?

**Comment by rcronin** [comment]
I'm not proud of this and not sure if this use case will work for everyone. I suppose it depends on what you're trying to do. I have a high order function being invoked in our next config to do some additional pre-compilations that needs to be run only once before the app builds/starts. I am checking for `process.argv[1].includes('.bin/next')`. I only run my pre-compilation if that is true.

Definitely not a fix...but a workaround.
